plugins/extractors/bigquery/bigquery.go

```diff
@@ -150,10 +150,9 @@ func (e *Extractor) extractTable(ctx context.Context, ds *bigquery.Dataset, emit
 	tb := ds.Tables(ctx)
 	for {
 		table, err := tb.Next()
-		if err == iterator.Done {
+		if errors.Is(err, iterator.Done) || errors.Is(err, context.Canceled) {
 			break
-		}
-		if err != nil {
+		} else if err != nil {
```

Comment:
`else` is not required here.

Reply:
it would take one extra line right if we remove `else`?

Reply:
Sometimes things are clearer with more lines of code. In this case, my intention was to keep it similar to the recommendation to avoid an else condition when we return from inside the if condition.

that said, don't have a strong opinion though, okay with it either way.

Reply:
valid concern, but in this exact case I think it is still fine.

will keep it like this for now to resolve this 🙏 .